assets/js/components/environments/SurveyPageNav.jsx

```diff
@@ -16,49 +11,56 @@ class SurveyPageNav extends Component {
   constructor(props) {
     super(props)
     this.state = {
-      tab: 1
+      tab: this.currentTab
     }
   }
 
-  handleSelect(tabIndex) {
-
-    this.setState({
-      tab: tabIndex
-    })
+  get currentTab() {
+    const current = this.routes.indexOf(window.location.pathname) + 1
+    return current > 0 ?
+      current :
+      this.activeTabIndex
+  }
 
-    let route
-    switch (tabIndex) {
-        case 1:
-          route = '/survey'
-          break
-        case 2:
-          route = '/survey/questions'
-          break
-        case 3:
-          route = '/survey/results'
-          break
-    }
+  get routes() {
+    return [ '/survey', '/survey/questions', '/survey/results' ]
+  }
 
-    browserHistory.push(route)
+  get activeTabIndex() {
+    return this.state ?
+      this.state.tab :
+      1
+  }
 
+  set activeTabIndex(index) {
+    this.setState({
+      tab: index
+    })
   }
 
   render() {
+    const tab = this.currentTab - 1,
+      [ward, survey, results] = this.routes.map((route, i) => 
+        i < tab ?
+          { href: route } :
```

Comment:
instead of using hrefs, the more SPA way that we should take advantage of is using React Router which is baked into our project. They suggest using the Navigator `transitionTo` method, this links to routes defined in the router as we have.

This will refrain from causing a full page refresh and preserve our app state if someone uses the breadcrumbs. 

Otherwise excellent work! I appreciate your tests with your components, i learned something!